### ISSUE TITLE
Can you make it to where you can export the models back into .bmd?

### ISSUE THREAD

**Issue by M3KAI5ER44** [niacdoial/blemd#29]
I installed this thinking it would go both ways... And I can't figure out how to use the other versions like BMDcubed or SuperBMD...

**Comment by M3KAI5ER44** [comment]
@niacdoial can you help me? Despite my best efforts, I cannot seem to find a way to convert the file format back to a .BMD, and I am having troubles with the tools in question. My project has been going slowly with many restless nights, and I would appreciate if you can point me in the right direction.

**Comment by niacdoial** [comment]
Hello,
Long story short, using BleMD as a way to write bmd files is a (very) long term goal. There is a lot of stuff I want to do before that, not the least of which is porting BleMD to blender 2.8, and rewrite the low-level parts of the program in a more human-readable and elegant way.
There are some functions aimed to export a model into the bmd format, but it is nowhere near completion, and some of these functions are half-written.

However, since I noticed you forked this project, I worked on a file that explains the structure of the program to anyone willing to tinker with it. (it should arrive with the incoming commit)

For BMDCubed and SuperBMD, I cannot use them myself, but it is because I haven't found a way to recompile them for my (linux) PC yet, so I won't be able to help you there.

**Comment by M3KAI5ER44** [comment]
Seeing how I have some free time I am going to figure out how to compile SuperBMD so I can finish my project then send I will the compiled version if I can. Thank you for responding.

**Comment by M3KAI5ER44** [comment]
I got the .exe and stuff and since you are using linux here is the link to WINE for running the program [https://www.winehq.org/](url) and the .bats can be ran in WINE too (As I've heard
[Release.zip](https://github.com/niacdoial/blemd/files/3318855/Release.Over.Here.zip)
)

**Comment by Sage-of-Mirrors** [comment]
Hello! Author of the original BMDCubed and SuperBMD here.

Do you still have plans to incorporate output of models and animations to the J3D formats (bmd/bdl/bck/etc)? I've been away from the scene for a while, but I'm starting to get some motivation to return. BleMD is light-years ahead of what SuperBMD could do, especially with respect to animations; SuperBMD could export the joint bindings correctly, and you could pose the meshes in Blender, but I could never figure out how to export the bindings back to COLLADA without fucking everything up. If you would have me, I'd like to help out by writing the export side of this plugin. It would be a huge leap forward for modding J3D games if developing custom models and animations was as simple as modeling in Blender and hitting an `export` button.

Hypothetically, once this is done, I could continue working on my C++ J3D rendering library. When that's in a usable state I could make an editor for configuring TEV materials; the workflow would then be `Model and rig a mesh -> export with BleMD -> import into material editor -> configure materials -> export finished mesh`.

Please let me know what you think!

**Comment by niacdoial** [comment]
Hello!

Adding an export functionality to BleMD was never really off the table, it's just that my motivation was… well, I guess you see the amount of commits that appeared recently. :')

Honestly, if you want to help, I'll be more than glad! <br/>
I believe I have a bunch of unused code that is supposed to start doing that (especially in vtx1, trying to naively decompose the mesh into triangle strips)

Don't be afraid of asking questions about what the code does, since it's probably not my most legible work.
(if you have a public key on hand, I can even send you a link to an account of mine that's better for 1-on-1 communication)

tl;dr: I think it's awesome!

**Comment by Sage-of-Mirrors** [comment]
Awesome! I'll fork the repo and take a look later tonight.

Public key? For what?

**Comment by niacdoial** [comment]
er… I thought I'ld send my username in an encrypted message that only you can read?
(that might be overkill though)

**Comment by Sage-of-Mirrors** [comment]
I apologize, I've never used any messaging system involving encryption keys. My handle is Gamma#1615 if you want to friend me real quick - when you do I'll edit this message to remove it.

**Comment by niacdoial** [comment]
here, removed the handle-containing version of this comment from history.
